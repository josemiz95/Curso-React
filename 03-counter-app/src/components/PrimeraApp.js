import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ nombre, subtitulo }) => {    
    	//Fragment es para evitar poner un div
    	let html = (
		<>
			<h1>Hola {nombre}</h1>
			<p>{subtitulo}</p>
		</>
	 );

	 return html;
}

// Validacion de tipo de objectos
PrimeraApp.propTypes = {
	nombre: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
	subtitulo: 'Este es el valor por defecto'
};

export {
	PrimeraApp as default
}