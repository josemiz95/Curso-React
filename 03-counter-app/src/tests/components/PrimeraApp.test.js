import React from 'react';
import { render } from "@testing-library/react";
import PrimeraApp from '../../components/PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp />', () =>{
    test('Debe mostrar hola Josemi', () =>{
		const nombre = 'Josemi';
		
		const wrapper = render(<PrimeraApp nombre={nombre}/>);

		expect(wrapper.getByText(`Hola ${nombre}`)).toBeInTheDocument();
	});
});

describe('Pruebas en <PrimeraApp /> con Enzyme', () =>{
	test('Debe mostrar hola Josemi', () =>{
		const nombre = 'Josemi';
		
		const wrapper = shallow(<PrimeraApp nombre={nombre}/>);

		expect(wrapper).toMatchSnapshot(); // para verificar con su snapshot
	});

	test('Debe mostrar subtitulo', () =>{
		const nombre = 'Josemi';
		const subtitulo = 'Este es un subtitulo';
		
		const wrapper = shallow(<PrimeraApp nombre={nombre} subtitulo={subtitulo}/>);
		const textoParrafo = wrapper.find('p'); // es igual que querySelector

		expect(textoParrafo.text()).toBe(subtitulo);
	});
});