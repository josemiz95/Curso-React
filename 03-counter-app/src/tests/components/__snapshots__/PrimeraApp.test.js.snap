// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> con Enzyme Debe mostrar hola Josemi 1`] = `
<Fragment>
  <h1>
    Hola 
    Josemi
  </h1>
  <p>
    Este es el valor por defecto
  </p>
</Fragment>
`;
